search/materialize.go

```diff
@@ -464,62 +582,206 @@ type valuesIterator struct {
 	buffer             []parquet.Value
 	currentBufferIndex int
 	err                error
+	ready              bool
+}
+
+// release releases any pooled resources held by the iterator, and should be
+// called when the iterator is no longer needed. Although helpful for efficient
+// memory management, it is not strictly necessary to call this method.
+func (pi *pageIterator) release() {
+	parquet.Release(pi.p)
 }
 
-func (vi *valuesIterator) Reset(p parquet.Page) {
-	vi.p = p
-	vi.vr = nil
+func (pi *pageIterator) readPage(pgs parquet.Pages) {
+	parquet.Release(pi.p)
+	p, err := pgs.ReadPage()
+	if err != nil {
+		pi.err = errors.Wrap(err, "failed to read page")
+		return
+	}
+	pi.p = p
+	pi.vr = nil
 	if p.Dictionary() != nil {
-		vi.st.Reset(p)
-		vi.cachedSymbols = make(map[int32]parquet.Value, p.Dictionary().Len())
+		pi.st.Reset(p)
+		pi.cachedSymbols = make(map[int32]parquet.Value, p.Dictionary().Len())
 	} else {
-		vi.vr = p.Values()
-		vi.buffer = make([]parquet.Value, 0, 128)
-		vi.currentBufferIndex = -1
+		pi.vr = p.Values()
+		pi.buffer = make([]parquet.Value, 0, 128)
```

Comment:
This was there before but wondering if 128 should be a default variable instead of a magic number here.

Reply:
You mean making it a constant at the top? Or making it configurable? I'd rather leave the latter for a followup PR

Reply:
I meant a constant at the top but it was a minor nit, its not something you added.